src/main/java/com/fauna/client/FeedIterator.java

```diff
@@ -0,0 +1,113 @@
+package com.fauna.client;
+
+
+import com.fauna.exception.FaunaException;
+import com.fauna.feed.FeedRequest;
+import com.fauna.feed.FeedSuccess;
+import com.fauna.response.StreamEvent;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+/**
+ * FeedIterator iterates over Event Feed pages from Fauna.
+ * @param <E>
+ */
+public class FeedIterator<E> implements Iterator<FeedSuccess<E>> {
+    private final FaunaClient client;
+    private final Class<E> resultClass;
+    private final FeedRequest initialRequest;
+    private CompletableFuture<FeedSuccess<E>> feedFuture;
+
+    /**
+     * Construct a new PageIterator.
+     * @param client            A client that makes requests to Fauna.
+     * @param request           The Feed Request object.
+     * @param resultClass       The class of the elements returned from Fauna (i.e. the rows).
+     */
+    public FeedIterator(FaunaClient client, FeedRequest request, Class<E> resultClass) {
+        this.client = client;
+        this.resultClass = resultClass;
+        this.initialRequest = request;
+        this.feedFuture = client.asyncFeed(request, resultClass);
+    }
+
+    @Override
+    public boolean hasNext() {
+        return this.feedFuture != null;
+    }
+
+    /**
+     * Returns a CompletableFuture that will complete with the next page (or throw a FaunaException).
+     * @return  A c
+     */
+    public CompletableFuture<FeedSuccess<E>> nextAsync() {
+        if (this.feedFuture != null) {
+            return this.feedFuture.thenApply(fs -> {
+                if (fs.hasNext()) {
+                    FeedRequest.Builder builder = FeedRequest.builder(initialRequest.getToken()).cursor(fs.getCursor());
+                    initialRequest.getPageSize().ifPresent(builder::pageSize);
+                    this.feedFuture = client.asyncFeed(builder.build(), resultClass);
+                } else {
+                    this.feedFuture = null;
+                }
+                return fs;
+            });
+        } else {
+            throw new NoSuchElementException();
+        }
+    }
```

Comment:
If I understand this code correctly we're returning the result of the already requested page while issuing the next page request in the background. If so, I think that's a bit counter intuitive to customers since they lose control on rather to call the next page or not.

Reply:
Eric and I chatted about this, I added `E2EFeedsTest.manualFeed` to explicitly show how a user can control the requests being sent to Fauna, but I left the implementation of `FeedIterator` as-is.